src/util/simplify_expr_floatbv.cpp

```diff
@@ -137,26 +137,21 @@ bool simplify_exprt::simplify_sign(exprt &expr)
 }
 #endif
 
-bool simplify_exprt::simplify_floatbv_typecast(exprt &expr)
+simplify_exprt::simplify_resultt
+simplify_exprt::simplify_floatbv_typecast(const floatbv_typecast_exprt &expr)
```

Comment:
Why is removing _this_ side effect an improvement?

Reply:
@peterschrammel The removal of the side effect is a side effect. The point is to allow us to pass a `floatbv_typecast_exprt` as the parameter rather than an `expr` - we can't mutate the type with the `floatbv_typecast_exprt` parameter though because that could change its type (and then we'd effectively be lying about the signature of the function).

Arguably something like that should be done for all the functions in here.